tests/chainer_tests/functions_tests/array_tests/test_pad_sequence.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'lengths': [[2, 1, 5, 3], [2]],
+    'length': [None, 6],
+    'shape': [(3, 4), ()],
+    'pad': [0, -1],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
```

Comment:
How about adding these cases?
* `lengths`:
  - `[]`, `[0]`
  - sequence including `0`
* `length`: exactly the same as `max(lengths)`
* `pad`: nan, inf
* `dtype`: non-float types (only forward computation)